### PR TITLE
Add ScalaFMT Config, guide & apply to projects. 

### PR DESCRIPTION
**Pull request by phillipbarron**
## What does this change?

Adds config for ScalaFMT and notes to the readme on its use via command line and automatically via the IDE. 
The PR also formats all three Scala projects

## How to test

There have been some linting changes applied as a result of the format. Removal of redundant enclosing blocks in case statements & formatted strings for example & using `nonEmpty` on lists rather than checking for `.size > 0`. 

All tests are passing however, spinning up the app and  verifying it is behaving as expected would be sensible.

### CODE REVIEW

**Review by jonathonherbert — APPROVED** [review]
This is great!

What do we think about running this in CI, and a pre-push githook (perhaps we could install it in the setup script?) – without enforcing these changes we'll drift away from scalafmt perfection.

Finally, shall we wait until #231 and #239 are merged before merging this? Suspect the conflicts will ... numerous.

**Comment by phillipbarron** [comment]
> 

I added some notes to the card - agree 100% with a pre-commit hook. There is an existing card to move from TeamCity to GHA which might add this, though there is nothing to stop us going that part right away

**Pull request merged**
